packages/dds/tree/src/shared-tree/treeCheckout.ts

```diff
@@ -758,14 +872,25 @@ export class TreeCheckout implements ITreeCheckoutFork {
 			"The source of the branch rebase has been disposed and cannot be rebased.",
 		);
 		assert(
-			!checkout.transaction.inProgress(),
+			!checkout.isTransacting(),
 			0x9af /* A view cannot be rebased while it has a pending transaction */,
 		);
 		assert(
 			checkout.isBranch,
 			0xa5d /* The main branch cannot be rebased onto another branch. */,
 		);
-		checkout._branch.rebaseOnto(this._branch);
+
+		const result = checkout._branch.rebaseOnto(this._branch);
+		if (result !== undefined && this.isTransacting()) {
+			const { targetCommits } = result.commits;
+			// If `targetCommits` were empty, then `result` would be undefined and we couldn't reach here
+			assert(hasSome(targetCommits), "Expected target commits to be non-empty");
+			const src = targetCommits[0].parent?.revision;
+			const dst = getLast(targetCommits).revision;
+			if (src !== undefined && dst !== undefined) {
```

Comment:
Would't be a bug if `src` were undefined?

Separately: `dst` can't be undefined here (according to the type system).

Reply:
I also don't know this code, but let's see... yea, if src doesn't have a parent, then that means that the rabase happened over the "root" commit, which is not part of the trunk, and which shouldn't be possible. So I think you are right. But, maybe one could argue that that's an implementation detail of EditManager, and this code is just being defensive and trying to handle the more general case of getting a commit's parent in a vacuum. We can make TreeCheckouts against branches without having an EditManager, and I think then this could happen?

Reply:
Well, I added an assert for `src` and all our tests pass, so - should be fine.